trie/pruning.go

```diff
@@ -214,12 +263,16 @@ func (p *pruner) loop() {
 // it, cascading until all dangling nodes are removed. If the pruner's interrupt
 // has been triggered (block processing pending), the remaining nodes are bubbled
 // up to the caller to reschedule later.
-func (p *pruner) prune(owner common.Hash, hash common.Hash, path []byte, tries []*traverser, batch ethdb.Batch) []*prunerTarget {
+func (p *pruner) prune(owner common.Hash, hash common.Hash, path []byte, taskset map[string]struct{}, tries []*traverser, batch ethdb.Batch) []*prunerTarget {
+	// If the node is already queued for pruning, don't duplicate any effort on it
+	key := makeNodeKey(owner, hash)
+	if _, ok := taskset[makeNodeKey(owner, hash)]; ok {
```

Comment:
use the `key`

Reply:
it can also be reused later, `dead := key` instead of `dead := []byte(makeNodeKey(owner, hash))` ... ?

Reply:
Ah yeah, I went back and forth a lot with this part, apparently left it in the worst condition :)) Good catch